### PR TITLE
Do not use unsupported URL API in the react-native API

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

React-Native cannot use the URL API. Unfortunately, most of the method available in the URL interface is unavailable (except some of them). Since the SubmissionUrl API is easy to implement manually, we can avoid using URL and use manual string manipulation to construct the URL.

This diff adjust the SubmissionUrl class to be re-usable in all our libraries

### CODE REVIEW

File: packages/sdk-core/src/model/http/SubmissionUrlInformation.ts [hunk 1]
```diff
@@ -80,7 +89,7 @@ export class SubmissionUrlInformation {
             // submit.backtrace.io/universe/token/format
             // by spliting the submission URL by `/` and dropping the last
             // part of the URL, the last element on the list is the token.
-            return submissionUrlParts[submissionUrlParts.length - 2];
+            return submissionUrlParts[submissionUrlParts.length - 2] ?? null;
         }
 
         const url = new URL(submissionUrl);
```
Comment (perf2711) [inline packages/sdk-core/src/model/http/SubmissionUrlInformation.ts:95]: You still use `URL` here.

File: packages/sdk-core/src/model/http/SubmissionUrlInformation.ts [hunk 2]
```diff
@@ -90,22 +99,33 @@ export class SubmissionUrlInformation {
...
-            pathParts[3] = desiredFormat;
-            url.pathname = pathParts.join('/');
+            pathParts[2] = desiredFormat;
+            return (
+                submissionUrl.substring(0, submitIndex + this.SUBMIT_PREFIX.length) +
+                pathParts.join('/') +
+                queryParameters
+            );
         } else {
+            const url = new URL(submissionUrl);
```
Comment (perf2711) [inline packages/sdk-core/src/model/http/SubmissionUrlInformation.ts:126]: You still use `URL` here.

File: packages/sdk-core/src/model/http/SubmissionUrlInformation.ts (hunk 1 above)
Comment (konraddysput) [reply packages/sdk-core/src/model/http/SubmissionUrlInformation.ts:95]: As I said - we're using only supported URL API. If something is unsupported I'm creating the URL manually.

File: packages/sdk-core/src/model/http/SubmissionUrlInformation.ts (hunk 2 above)
Comment (konraddysput) [reply packages/sdk-core/src/model/http/SubmissionUrlInformation.ts:126]: As I said - we're using only supported URL API. If something is unsupported I'm creating the URL manually.

**Review by perf2711 — APPROVED** [review]

**Pull request merged**